cocos/tween/tween.ts

```diff
@@ -176,22 +282,50 @@ export class Tween<T extends object = any> {
      * @method clone
      * @param target @en The target of clone tween @zh 克隆缓动的目标对象
      */
-    clone<U extends object = any> (target: U): Tween<U> {
+    clone<U extends object = any> (target?: U): Tween<U> {
         const action = this._union(false);
         const r = tween(target);
         return action ? r.insertAction(action) : r;
     }
 
     /**
      * @en
-     * Integrate all previous actions to an action.
+     * Integrate to an action by all previous actions or a range from the specific id to the last one.
      * @zh
-     * 将之前所有的 action 整合为一个 action。
+     * 将之前所有的动作或者从指定标识的动作开始的所有动作整合为一个顺序动作。
+     * @method union
+     * @param fromId @en The action with the specific ID to start integrating @zh 指定开始整合的动作标识
+     * @note @en Don't use -1 for id since -1 is the default value for internal actions
+     *       @zh 请不要使用 -1 作为标识，因为 -1 是内部动作标识的默认值
      */
-    union (): Tween<T> {
-        const action = this._union(false);
-        this._actions.length = 0;
-        if (action) this._actions.push(action);
+    union (fromId?: number): Tween<T> {
+        const unionAll = (): void => {
+            const action = this._union(false);
+            this._actions.length = 0;
+            if (action) this._actions.push(action);
+        };
+
+        if (fromId === undefined) {
+            unionAll();
+            return this;
+        }
+
+        const actions = this._actions;
+        const index = actions.findIndex((action) => action.getId() === fromId);
+        if (index === -1) {
```

Comment:
I think -1 is an invalid value, don't have to support it.

Reply:
Yes, add a warn log now.
```ts
const index = actions.findIndex((action) => action.getId() === fromId);
        if (index === -1) {
            warn(`union: found: ${invalidActionId}`);
            return this;
        } else {
```